src/main/java/seedu/souschef/logic/commands/SurpriseCommand.java

```diff
@@ -13,10 +25,43 @@
     public static final String MESSAGE_USAGE = COMMAND_WORD + ": SousChef will surprise you with a random recipe.\n"
             + "Example: " + COMMAND_WORD;
 
-    public static final String SHOWING_SURPRISE_MESSAGE = "SURPRISE!";
+    public static final String MESSAGE_SURPRISE_SUCCESS = "SURPRISE!";
+
+    private final Model model;
+    private Index randomIndex;
+    private Random rand = new Random();
+
+    public SurpriseCommand(Model model) {
+        this.model = model;
+    }
+
+    @Override
+    public CommandResult execute(History history) throws CommandException {
+        requireNonNull(model);
+
+        List<Recipe> filteredRecipeList = model.getFilteredList();
+
+        setRandomIndex();
+
+        if (randomIndex.getZeroBased() >= filteredRecipeList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_RECIPE_DISPLAYED_INDEX);
```

Comment:
Should not throw exception, instead u should run the generator again to get a new number.
Also throw parseException when the recipelist is empty, since nth can be displayed. CommandException is not caught anywhere right? If so, why throw it?